BLDC.py

```diff
@@ -17,21 +18,24 @@ def __init__(self, pwmPin, directionPin, frequency):
             self.pwmCtrl = PWM.Signal(pwmPin, frequency)
 
     def setDirection(self, direction):
+        self.stop()
+        time.sleep(0.2)
         if direction == 'CW':
             GPIO.write(self.directionPin, 'LOW')
         elif direction == 'CCW':
             GPIO.write(self.directionPin, 'HIGH')
         else:
             raise ValueError("Could not set motor direction, expected either: 'CW' or 'CCW'")
+        time.sleep(0.2)
 
     def setSpeed(self, speed):                     #speed in percentage
-        if speed < 50:
-            raise ValueError("Speed is too low, please set it to a value higher than 50%")
-        else:
-            self.speed = speed
+        self.stop()
+        self.speed = speed
+        time.sleep(0.2)
```

Comment:
can the speed be changed while the motor is running? if so, we should call `start()` again here.

same question for changing direction

Reply:
I think it's better to stop the motor before making a speed change. A change in direction is rough on a motor. The change in momentum messes with the gears.

Reply:
Don't think we should change speed direction while its moving. The setting should be independent. That's why it doesn't start again

Reply:
ok, so is the caller of these functions supposed to do:
```python
motor.stop()
motor.setSpeed(75)
motor.start()
```
or just
```python
motor.setSpeed(75)
motor.start()
```
If we expect them to call `start()` again, I think we should also expect them to call `stop()`. We can add an `assert self.speed = 0` to verify this.

For changing speed, what's the difference between changing from 0 to 75 and changing from 50 to 75? I feel like it's a common use case to change speed while moving?

Reply:
I guess the second block. My idea was that we want to ensure that the motor has stopped before changing speed/direction and we don't want to start right away in case the user is not ready for the immediate change. 

To answer your question, I was expecting Wally to always have a constant speed while following a path. So a change in speed would occur during setup (pre path planning). So if a user is testing out the speeds, every time the speed is changed Wally isn't running away from them, but I can change this if that's not what we planned.

Reply:
actually, I don't think we are even planning on surfacing speed to the user, so I'm not too concerned about that. I was just thinking about the general code/robot logic. I prefer the first block, but we can leave it as is.